src/earthdataVarinfo/handler.py

```diff
@@ -0,0 +1,70 @@
+import os
+import traceback
+import json
+import shutil
+from tempfile import mkdtemp
+
+from varinfo import VarInfoFromNetCDF4
+from varinfo.cmr_search import get_granules, get_granule_link, download_granule
+from varinfo.umm_var import export_all_umm_var_to_json, get_all_umm_var
+
+
+def main(event, context):
+  # Setup CMR search url from cmrRootUrl
+  cmr_url = os.environ['cmrRootUrl'] + '/search/'
+
+  temp_dir = None
+
+  # Pull the concept id and token from the body
+  collection_concept_id = event.get('conceptId')
+
+  # Get token and remove 'Bearer ' if present 
+  token = event.get('token')
+  
+  # These two arguments are required for varinfo, return an error if they are not provided
+  if collection_concept_id is None or token is None:
+    return {
+      'isBase64Encoded': False,
+      'statusCode': 500,
+      'body': {
+        'error': 'Collection Concept ID and Token are must be provided.'
+      }
+    }
+
+  try:
+    # Retrieve a list of 10 granules for the collection
+    granules = get_granules(collection_concept_id, cmr_env=cmr_url, token=token)
+
+    # Get the URL for the first granule (NetCDF-4 file):
+    granule_link = get_granule_link(granules)
+
+    # Make a temporary directory (this may cause permission issues?):
+    temp_dir = mkdtemp()
```

Comment:
Discussed on the call: [It's probably best to use a context manager](https://docs.python.org/3/library/tempfile.html#tempfile-examples), because that way the clean-up of the temporary file is handled for you. That would make it something like:

```
from tempfile import TemporaryDirectory


with TemporaryDirectory() as temp_dir:
    # Download file to lambda runtime environment
    local_granule = download_granule(granule_link, token, out_directory=temp_dir)

    # Parse the granule with VarInfo:
    var_info = VarInfoFromNetCDF4(local_granule)

    # Generate all the UMM-Var records:
    all_variables = get_all_umm_var(var_info)
```

You then won't need the `shutil.rmtree(temp_dir)`, because the context manager removes the temporary directory when you exit out of it (using the `__exit__` method).

Reply:
(Also, I think that clean-up would still happen if you hit an exception, which currently isn't the case as you hit the `return` in the `Except` block and then never get any further)

Reply:
Switched to context manager